HEN_HOUSE/egs++/shapes/egs_dynamic_shape/egs_dynamic_shape.h

```diff
@@ -0,0 +1,252 @@
+/*
+###############################################################################
+#
+#  EGSnrc egs++ dynamic shape headers
+#  Copyright (C) 2015 National Research Council Canada
+#
+#  This file is part of EGSnrc.
+#
+#  EGSnrc is free software: you can redistribute it and/or modify it under
+#  the terms of the GNU Affero General Public License as published by the
+#  Free Software Foundation, either version 3 of the License, or (at your
+#  option) any later version.
+#
+#  EGSnrc is distributed in the hope that it will be useful, but WITHOUT ANY
+#  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+#  FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for
+#  more details.
+#
+#  You should have received a copy of the GNU Affero General Public License
+#  along with EGSnrc. If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
+#
+#  Author:          Reid Townson
+#
+#  Contributors:
+#
+###############################################################################
+*/
+
+
+/*! \file egs_dynamic_shape.h
+ *  \brief A dynamic shape
+ *  \RT
+ */
+
+#ifndef EGS_DYNAMIC_SHAPE_
+#define EGS_DYNAMIC_SHAPE_
+
+#include "egs_shapes.h"
+#include "egs_rndm.h"
+#include "egs_application.h"
+
+#ifdef WIN32
+
+    #ifdef BUILD_DYNAMIC_SHAPE_DLL
+        #define EGS_DYNAMIC_SHAPE_EXPORT __declspec(dllexport)
+    #else
+        #define EGS_DYNAMIC_SHAPE_EXPORT __declspec(dllimport)
+    #endif
+    #define EGS_DYNAMIC_SHAPE_LOCAL
+
+#else
+
+    #ifdef HAVE_VISIBILITY
+        #define EGS_DYNAMIC_SHAPE_EXPORT __attribute__ ((visibility ("default")))
+        #define EGS_DYNAMIC_SHAPE_LOCAL  __attribute__ ((visibility ("hidden")))
+    #else
+        #define EGS_DYNAMIC_SHAPE_EXPORT
+        #define EGS_DYNAMIC_SHAPE_LOCAL
+    #endif
+
+#endif
+
+/*! \brief An dynamic shape
+
+\ingroup Shapes
+
+An dynamic shape is a shape that
+takes a random point from another shape and then
+applies a transformation, using a time sampling and interpolating between
+control points.
+
+An dynamic shape is defined using
+\verbatim
+:start shape:
+    library = egs_dynamic_shape
+    :start shape:
+        definition of the shape to be 'dynamic'
+    :stop shape:
+    :start motion:
+       control point 1 = time(1) x(1) y(1) z(1) xrot(1) yrot(1) zrot(1)
+       control point 2 = time(2) x(2) y(2) z(2) xrot(2) yrot(2) zrot(2)
+       .
+       .
+       .
+       control point N = time(N) x(N) y(N) z(N) xrot(N) yrot(N) zrot(N)
+    :stop motion:
+:stop source:
```

Comment:
Oops, @rtownson: I went ahead and approved the PR, but didn't see that this question wasn't answered.  I think this should end with ":stop shape:" instead of ":stop source:"  If that's the case, please make this change before merging.

Reply:
It's fixed!